### PR TITLE
WIP: Add Model, Refactor

### PR DESCRIPTION
**Pull request by mcmcgrath13**
* Add routes for the model in parallel to the historical buoy routes
* Change `utils` to `common` inside the `erddap/` directory, move intermediate functions here
* Make buoy routes barebones, referring instead to common
* Make model routes same as buoy
* Add erddap queries for buoy ids and variables (replaces hardcoded variables)
* Update queries to more efficiently query erddap (summary, multi-buoy)

### CODE REVIEW

File: app/init.js [hunk 1]
```diff
@@ -1,27 +1,21 @@
...
-
-  const payload = {
-    ids,
-    variable,
-    datasetId: 'combined_e784_bee5_492e',
-    start: '2000-07-01T12:00:00Z',
-    end
+  const keyBase = `__express__/erddap`;
+  const initMap = {
+    buoy: "combined_e784_bee5_492e",
```
Comment (mirestrepo) [inline app/init.js:9]: Maybe these initial buoy and model ids as well as the combined `DATASET_ID` in `buoy.js`  could be all abstracted together

File: app/routes/erddap/buoy.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
 const router = express.Router();
-const { getMultiBuoyGeoJsonData, getBuoysCoordinates } = require('@/clients/erddap');
-const utils = require('@/utils');
-const buoys = require('@/routes/erddap/utils');
-const { cacheMiddleware } = require('@/middleware/cache');
+const utils = require("@/utils");
+const common = require("@/routes/erddap/common");
+const { cacheMiddleware } = require("@/middleware/cache");
+
+const DATASET_ID = "combined_e784_bee5_492e";
```
Comment (mirestrepo) [inline app/routes/erddap/buoy.js:7]: I think I would called the variable `COMBINED_DATASET_ID` or something like that just to make it clear that it is the combined

File: app/routes/erddap/model.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
+const express = require("express");
+const router = express.Router();
+const utils = require("@/utils");
+const common = require("@/routes/erddap/common");
+const { cacheMiddleware } = require("@/middleware/cache");
+
+const DATASET_ID = "model_data_77bb_15c2_6ab3";
```
Comment (mirestrepo) [inline app/routes/erddap/model.js:7]: Same here, I think I would name the variable `MODEL_DATASET_ID`

**Review by mirestrepo — COMMENTED** [review]
From what I can tell, most of the changes help readability and abstract the code, which is great.

I have a question... `bouy.js` is basically identical to `model.js`. Is there no way to have a common file that simply substitutes the API route? That is the only difference as far as I can tell

**Comment by mcmcgrath13** [comment]
> From what I can tell, most of the changes help readability and abstract the code, which is great.
> 
> I have a question... `bouy.js` is basically identical to `model.js`. Is there no way to have a common file that simply substitutes the API route? That is the only difference as far as I can tell

Thanks for looking! Yeah, I'm thinking buoy/model can be combined into a common route with a matched path (`/erddap/:route/target`) which would simplify the two.  They might diverge in the future (especially model, when thinking about the gridded data), but I think that could then be handled with hardcoded paths.  I need to read up on expresses path matching rules to be sure.

**Pull request merged**